DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java

```diff
@@ -443,34 +361,9 @@ public SymbolTableType getSymbolTableType() {
         if (!CharSequence.class.isAssignableFrom(dataType) || !isDirect()) {
             return null;
         }
-        if (Objects.equals(isVarSizeString, Boolean.TRUE)) {
```

Comment:
This isn't the right change. We need to do one of two things:
1. Explicitly specify symbol table type or symbol table "source".
2. Stop "knowing" anything about symbol tables at the column source level.

Reply:
I'm OK with the first approach, as a boolean called "hasSymbolTable". (Renames and inverts the boolean).

Reply:
Take a look at current code/main; isVarSizeString is never used from column source (or anywhere other than generating the table definitions table).  It has no effect.

Reply:
`ColumnDefinition.getSymbolTableType()` is the important bit. Used in `com.illumon.iris.db.v2.sources.regioned.RegionedTableComponentFactoryImpl#createRegionedColumnSource` and other places (which you removed, but need to re-add).

Reply:
There must also be something in the writer that is sure to use a dictionary.